kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/exceptionhandlers/IgnoreRecordTooLargeHandler.java

```diff
@@ -0,0 +1,27 @@
+package org.hypertrace.core.kafkastreams.framework.exceptionhandlers;
+
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.errors.RecordTooLargeException;
+import org.apache.kafka.streams.errors.ProductionExceptionHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+public class IgnoreRecordTooLargeHandler implements ProductionExceptionHandler {
```

Comment:
Since we are internally handling multiple producer exceptions through if-else, let's rename the class to `DefaultKafkaProducerExceptionHandler`

Reply:
Also we need to register this class within KafkaStreamsApp.getBaseStreamsConfig():

```
baseStreamsConfig.put(DEFAULT_PRODUCTION_EXCEPTION_HANDLER_CLASS_CONFIG,
DefaultKafkaProducerExceptionHandler.class.getName());
```

Reply:
IMHO, we should not auto-configure this property by default. Let the user configure it as per service requirement. he/she should adjust value of property `max.record.size` accordingly.
Also, user can now specify multiple exceptions which can be ignored. By default, the list of exceptions is not defined and exception handler will return fail.